internal/resources/aws/glue_catalog_database.go

```diff
@@ -0,0 +1,112 @@
+package aws
+
+import (
+	"github.com/shopspring/decimal"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+)
+
+// GlueCatalogDatabase struct represents a serverless AWS Glue catalog. A Glue catalog is a database designed to store
+// raw data fetched from Glue crawlers before the data is cleaned and transformed by a Glue job.
+//
+// GlueCatalogDatabase is just one resource of the wider AWS Glue service, which provides a number of different serverless services
+// to build a robust data analytics pipeline.
+//
+// Resource information: https://aws.amazon.com/glue/
+// Pricing information: https://aws.amazon.com/glue/pricing/
+type GlueCatalogDatabase struct {
+	Address string
+	Region  string
+
+	MonthlyObjects  *float64 `infracost_usage:"monthly_objects"`
+	MonthlyRequests *float64 `infracost_usage:"monthly_requests"`
+}
+
+// PopulateUsage parses the u schema.UsageData into the GlueCatalogDatabase.
+// It uses the `infracost_usage` struct tags to populate data into the GlueCatalogDatabase.
+func (r *GlueCatalogDatabase) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+// BuildResource builds a schema.Resource from a valid GlueCatalogDatabase struct. GlueCatalogDatabase has the following
+// schema.CostComponents associated with it:
+//
+//		1. Storage - charged for every 100,000 objects stored above 1M, per month.
+//		2. Requests - charged per million requests above 1M in a month.
+//
+// This method is called after the resource is initialised by an IaC provider. See providers folder for more information.
+func (r *GlueCatalogDatabase) BuildResource() *schema.Resource {
+	return &schema.Resource{
+		Name: r.Address,
+		UsageSchema: []*schema.UsageItem{
+			{Key: "monthly_objects", DefaultValue: 0, ValueType: schema.Float64},
+			{Key: "monthly_requests", DefaultValue: 0, ValueType: schema.Float64},
+		},
+		CostComponents: []*schema.CostComponent{
+			r.storageObjectsCostComponent(),
+			r.requestsCostComponent(),
+		},
+	}
+}
+
+func (r GlueCatalogDatabase) storageObjectsCostComponent() *schema.CostComponent {
+	var quantity *decimal.Decimal
+	var limit float64 = 100000
+
+	if r.MonthlyObjects != nil {
+		objects := *r.MonthlyObjects
+
+		if objects < limit {
+			objects = limit
+		}
```

Comment:
If the number of objects is less than free limit, should this cost component be free? Otherwise, we'll need to pay for one unit.

Reply:
Yeah I wasn't sure about this as we normally discount the free tier across resources, but thats because we don't know if the baseline free limit has already been reached. In this case we have all the information required - so it makes sense. I'll fixup